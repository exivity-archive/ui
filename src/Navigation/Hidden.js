import PropTypes from 'prop-types'
import { styled, keyframes, Hidden as BaseHidden } from 'reakit'
import { prop } from 'styled-tools'

const bounce = keyframes`
  0%, 50%, 75%, 87.5%, 100% { transform: translateY(0) }
  30% { transform: translateY(-100%) }
  65% { transform: translateY(-50%) }
  82.5% { transform: translateY(-25%) }
`

const rotateAndSlide = keyframes`
  0% { transform: none }
  50% { transform: rotateZ(180deg) }
  100% { transform: rotateZ(180deg) translateX(1000%) }
`

const Hidden = styled(BaseHidden)`
  background: palevioletred;
  width: 20px;
  height: 20px;
  margin: 10px;
  &[aria-hidden="false"] {
    animation: ${bounce} ${prop('duration')};
  }
  &[aria-hidden="true"] {
    animation: ${rotateAndSlide} ${prop('duration')};
  }
`

Hidden.propTypes = {
  ...BaseHidden.propTypes,
  children: PropTypes.node
}

Hidden.defaultProps = {}

Hidden.displayName = 'Navigation.Hidden'

export default Hidden
