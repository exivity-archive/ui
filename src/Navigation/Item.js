import PropTypes from 'prop-types'
import { styled, Box } from 'reakit'
import { theme } from 'styled-tools'

const Item = styled(Box.as('li'))`
  ${theme('NavigationItem')};
  
  list-style: none;
`

Item.propTypes = {
  ...Box.propTypes,
  children: PropTypes.node
}

Item.defaultProps = {}

Item.displayName = 'Navigation.Item'

export default Item
