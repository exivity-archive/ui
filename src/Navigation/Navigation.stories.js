import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import { Hidden, Link } from 'reakit'
import {
  MdDashboard,
  MdInsertChart,
  MdLibraryBooks
} from 'react-icons/md'

import Navigation from './Navigation'
import Icon from './../Icon'

storiesOf('molecules|Navigation', module)
  .add('item', () => <Navigation.Item>
    <Link href='.'>
      <Icon><MdDashboard /></Icon>
      Dashboard
    </Link>
  </Navigation.Item>)
  .add('toggle', () => <Navigation.Toggle toggle={action('toggle')}>
    <Icon><MdDashboard /></Icon>
    Dashboard
  </Navigation.Toggle>)
  .add('example', () => <Navigation>
    <Navigation.Item>
      <Link href='.'>
        <Icon><MdDashboard /></Icon>
        Dashboard
      </Link>
    </Navigation.Item>
    <Navigation.Item>
      <Link href='.'>
        <Icon><MdInsertChart /></Icon>
        Reports
      </Link>
    </Navigation.Item>
    <Navigation.Item>
      <Link href='.'>
        <Icon><MdLibraryBooks /></Icon>
        Catalogue
      </Link>
    </Navigation.Item>
  </Navigation>)
  .add('toggled', () => <Navigation>
    <Hidden.Container>
      {hidden => (
        <Navigation.Item>
          <Navigation.Toggle {...hidden}>
            <Icon><MdDashboard /></Icon>
            Dashboard
          </Navigation.Toggle>
          <Navigation.Hidden {...hidden}>
            <Navigation.Item>
              <Link href='.'>Accounts</Link>
            </Navigation.Item>
            <Navigation.Item>
              <Link href='.'>Services</Link>
            </Navigation.Item>
            <Navigation.Item>
              <Link href='.'>Instances</Link>
            </Navigation.Item>
          </Navigation.Hidden>
        </Navigation.Item>
      )}
    </Hidden.Container>
    <Navigation.Item>
      <Link href='.'>
        <Icon><MdInsertChart /></Icon>
        Reports
      </Link>
    </Navigation.Item>
    <Navigation.Item>
      <Link href='.'>
        <Icon><MdLibraryBooks /></Icon>
        Catalogue
      </Link>
    </Navigation.Item>
  </Navigation>)
