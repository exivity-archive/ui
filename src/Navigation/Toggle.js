import React from 'react'
import PropTypes from 'prop-types'
import { styled, Hidden, Link } from 'reakit'
import { MdExpandMore, MdExpandLess } from 'react-icons/md'

import Icon from './../Icon'
import Item from './Item'

const ExpandIcon = () => <Icon><MdExpandMore /></Icon>
const CollapseIcon = () => <Icon><MdExpandLess /></Icon>

const StyledHiddenToggle = styled(Hidden.Toggle)`
  cursor: pointer;
`

const Toggle = ({ children, visible, ExpandIcon, CollapseIcon, ...otherProps }) => {
  return <StyledHiddenToggle
    as={[Link, 'button']}
    visible={visible}
    {...otherProps}>
    {/* workaround for bug https://codepen.io/rachelandrew/pen/dzXwRJ */}
    <Link as={[Item, 'div']} {...otherProps}>
      {children}
      {visible ? <CollapseIcon /> : <ExpandIcon />}
    </Link>
  </StyledHiddenToggle>
}

Toggle.propTypes = {
  ...Hidden.Toggle.propTypes,

  children: PropTypes.node,

  ExpandIcon: PropTypes.func,
  CollapseIcon: PropTypes.func
}

Toggle.defaultProps = {
  ExpandIcon,
  CollapseIcon
}

Toggle.displayName = 'Navigation.Toggle'

export default Toggle
