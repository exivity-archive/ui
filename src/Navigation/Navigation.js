import React from 'react'
import PropTypes from 'prop-types'
import { Navigation as BaseNavigation, styled } from 'reakit'

import List from './../List'

import Item from './Item'
import Toggle from './Toggle'
import Hidden from './Hidden'

const NavigationWrapper = styled(BaseNavigation)``

const Navigation = ({ children, ...otherProps }) => {
  return <NavigationWrapper {...otherProps}>
    <List>
      {children}
    </List>
  </NavigationWrapper>
}

Navigation.Item = Item
Navigation.Toggle = Toggle
Navigation.Hidden = Hidden

Navigation.propTypes = {
  ...BaseNavigation.propTypes,
  children: PropTypes.node
}

Navigation.defaultProps = {}

Navigation.displayName = 'Navigation'

export default Navigation
