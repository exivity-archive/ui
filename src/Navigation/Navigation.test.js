/* eslint-disable no-undef */
import React from 'react'
import renderer from 'react-test-renderer'

import Navigation from './Navigation'

test('renders default Navigation', () => {
  const component = renderer.create(<Navigation />)
  expect(component.toJSON()).toMatchSnapshot()
})
