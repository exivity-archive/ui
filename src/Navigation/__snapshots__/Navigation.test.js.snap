// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default Navigation 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

<nav
  className="c0"
>
  <ul
    className="c0"
  />
</nav>
`;
